tests/rptest/e2e_tests/flink_scale_test.py

```diff
@@ -0,0 +1,135 @@
+
+
+class FlinkScaleTests(RedpandaTest):
+    def __init__(self, test_context, *args, **kwargs):
+        # Init parent
+        super(FlinkScaleTests, self).__init__(test_context, log_level="trace")
+
+        self.flink = FlinkService(test_context)
+        # Prepare client
+        config = self.redpanda.security_config()
+        user = config.get("sasl_plain_username")
+        passwd = config.get("sasl_plain_password")
+        protocol = config.get("security_protocol", "SASL_PLAINTEXT")
+        self.kafkacli = KafkaCliTools(self.redpanda,
+                                      user=user,
+                                      passwd=passwd,
+                                      protocol=protocol)
+        # Prepare Workloads
+        self.workload_manager = WorkloadManager(self.logger)
+
+        return
+
+    def tearDown(self):
+        # Clean all up
+        for spec in self.topic_specs:
+            self.kafkacli.delete_topic(spec.name)
+        return super().tearDown()
+
+    def _run_workloads(self, workloads, config):
+        """
+            Run workloads from the list with supplied config
+
+            Return: list of failed jobs
+        """
+        # Run produce job
+        for workload in workloads:
+            # Add script as a job
+            self.logger.info(f"Adding {workload['name']} to flink")
+            ids = self.flink.run_flink_job(workload['path'], config)
+            if ids is None:
+                raise RuntimeError("Failed to run job on flink for "
+                                   f"workload: {workload['name']}")
+
+            self.logger.debug(f"Workload '{workload['name']}' "
+                              f"generated {len(ids)} "
+                              f"jobs: {', '.join(ids)}")
+
+        return ids
+
+    @cluster(num_nodes=4)
+    def test_transactions_scale(self):
+        """
+            Test uses same workload with different modes to produce
+            and consume/process given number of transactions
+        """
+        def assert_failed_jobs(job_list):
+            # Collect failed jobs
+            failed = []
+            for id in job_list:
+                job = self.flink.get_job_by_id(id)
+                if job['state'] == self.flink.STATE_FAILED:
+                    self.logger.warning(f"Job '{id}' has failed")
+                    failed.append(job)
+
+            desc = [f"{j['name']} ({j['jid']}): {j['state']}" for j in failed]
+            desc = "\n".join(desc)
+            assert len(failed) == 0, \
+                "Flink reports fails for " \
+                f"transaction workload:\n{desc}"
+
+        # TODO: Validate/Gather info on
+        # - Transactions can be generated per one flink job
+        # - Transactions per node
+        # - Transactions total >500K
+
+        # Prepare topics
+        self.topic_name = "flink_scale_1"
+        self.topic_specs = [TopicSpec(name=self.topic_name, partition_count=8)]
```

Comment:
there is a single topic, remove list?

Reply:
Idea is that next test from this class will have multiple topics for each workload created and this is why the is a list. Originally, it had 4 different topics and after discussing it - there is one. Topic removal happens in tearDown